File: baselines/go/main-fibonacci.go
package main

import (
	"context"
	"fmt"

	"github.com/aws/aws-lambda-go/lambda"
)

// Fibonacci recursive implementation
// Source: ruchy-book bench-007-fibonacci.go
func fibonacci(n int) int {
	if n <= 1 {
		return n
	}
	return fibonacci(n-1) + fibonacci(n-2)
}

type testResponse struct {
	StatusCode int    `json:"statusCode"`
	Body       string `json:"body"`
}

func handleRequest(ctx context.Context) (testResponse, error) {
	// Calculate fibonacci(35) - standard Lambda benchmark
	result := fibonacci(35)

	return testResponse{
		StatusCode: 200,
		Body:       fmt.Sprintf("fibonacci(35)=%d", result),
	}, nil
}

func main() {
	lambda.Start(handleRequest)
}

File: baselines/rust/src/main-fibonacci.rs
use lambda_runtime::{service_fn, LambdaEvent, Error};
use serde_json::{json, Value};

// Fibonacci recursive implementation
// Source: ruchy-book bench-007-fibonacci pattern
fn fibonacci(n: i32) -> i32 {
    if n <= 1 {
        n
    } else {
        fibonacci(n - 1) + fibonacci(n - 2)
    }
}

#[tokio::main]
async fn main() -> Result<(), Error> {
    let func = service_fn(func);
    lambda_runtime::run(func).await?;
    Ok(())
}

async fn func(_: LambdaEvent<Value>) -> Result<Value, Error> {
    // Calculate fibonacci(35) - standard Lambda benchmark
    let result = fibonacci(35);

    Ok(json!({
        "statusCode": 200,
        "body": format!("fibonacci(35)={}", result)
    }))
}
